Mu documentation changed - New section "Generate sources from IDL" (#621)
higherkindness/mu: docs/src/main/tut/core-concepts.md

First things first, the main difference with respect to [gRPC] is that [mu] doesn’t need `.proto` files, but can still use protobuf thanks to the [PBDirect] library, which allows it to read and write Scala objects directly to protobuf with no `.proto` file definitions. In summary:

* Your protocols, both messages and services, will reside with your business-logic in your Scala files using [scalamacros] annotations to set them up. We’ll see more details on this shortly.
* As a optional feature, [mu] can generate the services and messages definitions in Scala code from `.proto` files.  We'll check out this feature further in [this section](generate-sources-from-idl).

> * Also, instead of reading `.proto` files to set up the [RPC] messages and services, [mu] offers the ability to generate them based on the protocols defined in your Scala code. However, this feature is now **deprecated**, in future versions it will disapear.

Reviewer: ```suggestion
* With [mu] you can still generate `.proto` files based on the protocols defined in your Scala code. However, this feature is now **deprecated** and it will disappear in future versions.
```
